SConstruct

```diff
@@ -690,7 +553,212 @@ config_options = [
+            logger.info(config.list(), print_level=False)
+            sys.exit(0)
+
+        if defaults:
+            try:
+                logger.info(config.help(option), print_level=False)
+                sys.exit(0)
+            except KeyError as err:
+                message = "Run 'scons help --list-options' to see available options."
+                logger.error(f"{err}.\n{message}")
+                sys.exit(1)
+
+        dev = GetOption("dev") is not None
+        try:
+            message = config.to_rest(option, dev=dev)
+        except KeyError as err:
+            message = "Run 'scons help --list-options' to see available options."
+            logger.error(f"{err}.\n{message}")
+            sys.exit(1)
+
+        output = GetOption("output")
+        if output:
+            output_file = Path(output).with_suffix(".rst")
+            with open(output_file, "w+") as fid:
+                fid.write(message)
+
+            logger.info(f"Done writing output options to '{output_file}'.",
+                        print_level=False)
+
+        else:
+            logger.info(message, print_level=False)
+
+        sys.exit(0)
+
+# **************************************
+# *** Read user-configurable options ***
+# **************************************
+
+opts = Variables("cantera.conf")
+
+extraEnvArgs = {}
+
+if os.name == "nt":
+    config.add(windows_options)
+    config.add(config_options)
+
+    config["prefix"].default = pjoin(os.environ["ProgramFiles"], "Cantera")
+    config.select("Windows")
+
+    # On Windows, target the same architecture as the current copy of Python,
+    # unless the user specified another option.
+    if "64 bit" not in sys.version:
+        config["target_arch"].default = "x86"
+
+    opts.AddVariables(*config.to_scons(["msvc_version", "target_arch"]))
+
+    windows_compiler_env = Environment()
+    opts.Update(windows_compiler_env)
+
+    # Make an educated guess about the right default compiler
+    if which("g++") and not which("cl.exe"):
+        config["toolchain"].default = "mingw"
+
+    if windows_compiler_env["msvc_version"]:
+        config["toolchain"].default = "msvc"
+
+    opts.AddVariables(*config.to_scons("toolchain"))
+    opts.Update(windows_compiler_env)
+
+    if windows_compiler_env["toolchain"] == "msvc":
+        toolchain = ["default"]
+        if windows_compiler_env["msvc_version"]:
+            extraEnvArgs["MSVC_VERSION"] = windows_compiler_env["msvc_version"]
+        msvc_version = (windows_compiler_env["msvc_version"] or
+                        windows_compiler_env["MSVC_VERSION"])
+        logger.info(f"Compiling with MSVC {msvc_version}", print_level=False)
+
+    elif windows_compiler_env["toolchain"] == "mingw":
+        toolchain = ["mingw", "f90"]
+        extraEnvArgs["F77"] = None
+        # Next line fixes http://scons.tigris.org/issues/show_bug.cgi?id=2683
+        extraEnvArgs["WINDOWS_INSERT_DEF"] = 1
```

Comment:
This appears to be fixed now, see: https://github.com/SCons/scons/issues/2856 At the least, the link should be updated to the copied GitHub issue: https://github.com/scons/scons/issues/2683